Maple2.Server.Game/PacketHandlers/UgcHandler.cs

```diff
@@ -170,48 +225,111 @@ private void HandleConfirmation(GameSession session, IByteReader packet) {
         string ugcGuid = packet.ReadUnicodeString();
         packet.ReadShort(); // -255
 
-
-        Item? item = session.StagedUgcItem;
-        if (item?.Template == null || !TableMetadata.UgcDesignTable.Entries.TryGetValue(item.Id, out UgcDesignTable.Entry? ugcMetadata)) {
+        if (info.AccountId != session.AccountId || info.CharacterId != session.CharacterId || ugcUid == 0) {
+            Logger.Warning("Invalid UGC confirmation {AccountId} {CharacterId} {UgcUid}", info.AccountId, info.CharacterId, ugcUid);
             return;
         }
 
         using WebStorage.Request webRequest = WebStorage.Context();
         UgcResource? resource = webRequest.GetUgc(ugcUid);
-        if (resource == null) {
+        if (resource == null || resource.Id != ugcUid) {
+            Logger.Warning("Failed to find UGC resource {UgcUid}", ugcUid);
             return;
         }
-        item.Template.Url = resource.Path;
 
-        using GameStorage.Request gameRequest = session.GameStorage.Context();
-        item = gameRequest.CreateItem(session.CharacterId, item);
-        if (item == null) {
-            Logger.Fatal("Failed to create UGC Item {ugcUid}", ugcUid);
-            throw new InvalidOperationException($"Fatal: UGC Item creation: {ugcUid}");
+        switch (info.Type) {
+            case UgcType.Item or UgcType.Furniture or UgcType.Mount:
+                Item? item = session.StagedUgcItem;
+                if (item?.Template == null || !TableMetadata.UgcDesignTable.Entries.TryGetValue(item.Id, out UgcDesignTable.Entry? ugcMetadata)) {
+                    return;
+                }
+
+                item.Template.Url = resource.Path;
+
+                using (GameStorage.Request gameRequest = session.GameStorage.Context()) {
+                    item = gameRequest.CreateItem(session.CharacterId, item);
+                    if (item == null) {
+                        Logger.Fatal("Failed to create UGC Item {ugcUid}", ugcUid);
+                        throw new InvalidOperationException($"Fatal: UGC Item creation: {ugcUid}");
+                    }
+                }
+
+                if (!session.Item.Inventory.CanAdd(item)) {
+                    session.Item.MailItem(item);
+                    return;
+                }
```

Comment:
Is this return intended? It'll never send the UgcPackets below

Reply:
not sure how mail actually works , let me try to replicate and see what happens